compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -3570,7 +3570,8 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
       MatchResult.Stuck
 
     def recur(remaining: List[MatchTypeCaseSpec]): Type = remaining match
-      case (cas: MatchTypeCaseSpec.LegacyPatMat) :: _ if sourceVersion.isAtLeast(SourceVersion.`3.4`) =>
+      case (cas: MatchTypeCaseSpec.LegacyPatMat) :: _ if sourceVersion.isAtLeast(SourceVersion.`3.4`)
+          && (ctx.phaseId < ctx.base.elimOpaque.id || !ctx.base.elimOpaque.exists) =>
```

Comment:
Er ... I'm very uneasy about it. We should only do this as a very last resort. As is, we might (and probably will) reduce a match type differently before and after `elimOpaque`. IMO it's better to actually crash than to end up in that situation.

Reply:
At the very least, this should say *workaround* for #19434 with a big fat HACK/FIXME comment. It's not a real fix.

Reply:
I don't know what the right fix is.  We have an applied match type alias and the compiler is asked to reduce it some time later.  But now it's a plain alias, so it's "illegal".  What should we do?  In this case we don't reduce, and I'm happy with that.  We just need to find where to tell it not to.